### PR TITLE
refactor(CLI11): use submodule instead of plain source code (Resolves #15)

### LINKED ISSUE #15: Proper external library handling
Include external libraries (e.g. CLI11, easylogging++) properly => sub-moduled or prepared by the users.
[optional] create CMake files.

### PR DESCRIPTION
**Pull request by jo-bru**
For now, the simplest and cleanest solution seems to be including external projects as submodules.
By using this approach, the source files of the external libraries must be manually included in the Makefile. This can be simplified in the future by using Cmake.

I thought this might be a bit simpler than describing how to get the source code and defining the corresponding location.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,17 +2,19 @@ CXX = g++
 
 BUILD_DIR = ./build/
 INC_DIR = ./include/
+INC_LIB = ./lib/
```
Comment (iomz) [inline Makefile:5]: What are the underlying semantics of `INC_DIR` and `INC_LIB`? (Also `LIB_DIR`)

Comment (iomz) [reply Makefile:5]: Sorry for being picky!

**Review by iomz — COMMENTED** [review]
Have a question in Makefile.
The rest looks good to me – I think the simplest is to write an instruction though 😄

File: Makefile (hunk 1 above)
Comment (jo-bru) [reply Makefile:5]: `INC_DIR` holds the path to the include directory. As far as I understood it, this is would hold public headers, but as the xarm-commander won't be used as a library, I guess this folder won't be even used..
`INC_LIB` holds the path to the /lib directory which holds all external libraries (not sure if /external should be used for such submodules like the CLI11)
`LIB_DIR` holds the path to the build folder of the xarm SDK which includes the compiled binaries for the xarm-SDK
So the `INC_DIR` may be removed in future versions..

Comment (iomz) [reply Makefile:5]: According to the following sources:
- https://raw.githubusercontent.com/vector-of-bool/pitchfork/develop/data/spec.bs
- https://softwareengineering.stackexchange.com/questions/81899/how-should-i-organize-my-source-tree

`./lib` should then be `./libs` and `$LIB`, `INC_DIR` can go, and `LIB_DIR` can be more specific `XARM_SDK_LIB` or something?

Comment (jo-bru) [reply Makefile:5]: Yes, you're right!
I'll change that :)

**Review by iomz — APPROVED** [review]
Thank you @jo-bru !

**Comment by iomz** [comment]
lol the check failed 😂

**Comment by jo-bru** [comment]
yes, I'm having trouble with renaming the lib dir, because it includes submodules..but I'll get there 😂

**Comment by jo-bru** [comment]
finally..😂 sorry for the mess

**Pull request merged**